js/group_email.js

```diff
@@ -0,0 +1,195 @@
+            emailAddresses[emailAddresses.length] = result[i][3];
+        }
+        document.getElementById("number_label").innerHTML = "This group currently contains " + (emailAddresses.length) + " people.";
+    });
+}
+
+let draftId, draftMessage;
+function loadDraft() {
+    let subject = document.getElementById("input_subject").value;
+    sa.getDraftBySubject(subject).then(res => {
+        let draft = sa.parseDraftBySubject(res);
+        if (draft === null) {
+            alert("No draft found! Please check the subject line.");
+        } else if (draft.length > 1) {
+            alert("More than one draft found! Please use a more precise subject line.");
+        } else {
+            draft.then(res => {
+                draftId = res.result.id;
+                let raw = res.result.message.raw;
+                draftMessage = sa.decode(raw);
+                displayContent(draftMessage);
+                document.getElementById("preview").classList.remove("invisible");
+                document.getElementById("template").classList.add("invisible");
+                stepNumber = 3;
+            });
+        }
+    })
+}
+
+let draftSubject;
+function displayContent(msg) {
+    draftSubject = msg.substr(msg.indexOf("Subject: ") + 9, msg.indexOf("From") - msg.indexOf("Subject: ") - 9).trim();
+    document.getElementById("email_subject").innerHTML = draftSubject;
+    let startIndex = msg.indexOf("UTF-8") + 10;
+    let content = msg.substr(startIndex);
+    let endIndex = content.indexOf("--");
+    document.getElementById("preview_content").value = content.substr(0,endIndex);
+}
+
+function sendEmails() {
+    for (let i = 0; i < emailAddresses.length; i++) {
+        let finalMessage = sa.replaceVar(draftMessage, {"name":names}, i);
+        finalMessage = sa.addAddress(finalMessage, emailAddresses[i]);
+        finalMessage = window.btoa(finalMessage).replace(/\+/g, '-').replace(/\//g, '_').replace(/=+$/, '');
+        sa.sendEmail(finalMessage).then(res => {
+            if (i === emailAddresses.length - 1) {
+                sa.getTableHeaders("INTERACTION_TRACKING").then(res => {
+                    let headers = sa.parseTableHeaders(res);
+                    let toInsert = [headers];
+                    for (let i = 0; i < studentIds.length; i++) {
+                        toInsert[toInsert.length] = [studentIds[i], "Email", "Subject: " + draftSubject.trim(), getCurrentDate(), "TRUE"];
+                    }
+                    sa.insertIntoTableColValues(headers, "INTERACTION_TRACKING", sa.arrayToObjects(toInsert)).then(res => {
+                        if (sa.parseInsert(res) !== studentIds.length) {
+                            console.log("Something is wrong!");
+                        } else {
+                            alert("Emails sent successfully!");
+                            console.log(draftSubject);
+                            //window.location.href = "main_page.html";
```

Comment:
Why is this now commented out? It's weird to stay on the same page after emails have been sent.